src/features/HelpAndKnowledge/SolutionsHelp/SolutionDetails/Solutions/Generic/_components/AddContractorForm/index.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState } from 'react';
+import { Controller, FormProvider, useForm } from 'react-hook-form';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  Button,
+  Fieldset,
+  Form,
+  FormGroup,
+  Label,
+  TextInput
+} from '@trussworks/react-uswds';
+import {
+  MtoCommonSolutionKey,
+  useCreateMtoCommonSolutionContractorMutation
+} from 'gql/generated/graphql';
+import GetMTOSolutionContacts from 'gql/operations/ModelToOperations/GetMTOSolutionContacts';
+
+import Alert from 'components/Alert';
+import useMessage from 'hooks/useMessage';
+import useModalSolutionState from 'hooks/useModalSolutionState';
+
+type FormValues = {
+  contractorTitle: string;
+  contractorName: string;
+};
+
+const AddContractorForm = ({ closeModal }: { closeModal: () => void }) => {
+  const { t: contractorT } = useTranslation('mtoCommonSolutionContractor');
+  const { t: miscT } = useTranslation('mtoCommonSolutionContractorMisc');
+
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      contractorTitle: '',
+      contractorName: ''
+    },
+    mode: 'onChange'
+  });
+
+  const {
+    control,
+    handleSubmit,
+    reset,
+    formState: { isValid }
+  } = methods;
+
+  const { selectedSolution } = useModalSolutionState();
+  const { showMessage } = useMessage();
+  const [create] = useCreateMtoCommonSolutionContractorMutation({
+    refetchQueries: [
+      {
+        query: GetMTOSolutionContacts
+      }
+    ]
+  });
+  const [hasMutationError, setHasMutationError] = useState(false);
+
+  if (!selectedSolution) {
+    return null;
+  }
+
+  const onSubmit = (formData: FormValues) => {
+    create({
+      variables: {
+        key: selectedSolution.key.toUpperCase() as MtoCommonSolutionKey,
+        contractorTitle: formData.contractorTitle,
+        contractorName: formData.contractorName
+      }
+    })
+      .then(response => {
+        if (!response?.errors) {
+          showMessage(
+            <Trans
+              i18nKey={miscT('addContractor.success')}
+              values={{
+                contractor: formData.contractorName
+              }}
+              components={{
+                bold: <span className="text-bold" />
+              }}
+            />
+          );
+          closeModal();
+        }
+      })
+      .catch(() => {
+        setHasMutationError(true);
+      });
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <Form
+        className="maxw-none"
+        data-testid="mailbox-form"
+        id="mailbox-form"
+        onSubmit={handleSubmit(onSubmit)}
+      >
+        {hasMutationError && (
+          <Alert
+            type="error"
+            slim
+            headingLevel="h1"
+            className="margin-bottom-2"
+          >
+            {miscT('addContractor.error')}
+          </Alert>
+        )}
+        <Fieldset disabled={!selectedSolution}>
```

Comment:
I saw this on the other forms and not sure how I feel about how we should implement this.  Above on line 57 you are returning null for the entire component if `selectedSolution` is falsy.  This disabled check will never actually hit. I think this is fine but just wanted to point this out in case we wanted to sync on a way to standardize/handle if an incorrect/not found solution is mounted

Reply:
Yeah I feel the same! I tested with if we hit url with none existing solution, it got redirected. So it's not the best look, but since it's not really being hit at all, I thought it's okay to leave it like this, until future changes. 
Do you have other ideas? im all ears!